### DISCUSSION TITLE
Embedding Video

### DISCUSSION THREAD

**Discussion by mikepianka** [thuliteio/doks#237] [Q&A]
What's the best practice for embedding video? 

There is a shortcode for images like this:
`{{< img src="avocado.png" alt="avocado" caption="<em>A Tasty Avocado</em>" class="border border-dark" >}}`

Is there something equivalent to embed a video with responsive sizing? Using something like below does not scale the size of the video properly.

`<pre><div class="embed-responsive"><video autoplay loop muted playsinline><source src="pan.webm" type="video/webm"><source src="pan.mp4" type="video/mp4"></video></div></pre>`

**Comment by mikepianka** [comment]
Actually, just a bare video tag like this seems to work pretty well on its own. 

`<video width="100%" autoplay loop muted playsinline><source src="pan.webm" type="video/webm"><source src="pan.mp4" type="video/mp4"></video>`

**Reply by h-enk** [reply]
See https://github.com/h-enk/doks/issues/241

**Comment by h-enk** [comment]
BTW, Hugo also provides built-in shortcodes for [Vimeo ](https://gohugo.io/content-management/shortcodes/#vimeo)+ [YouTube](https://gohugo.io/content-management/shortcodes/#youtube). 

I will add the video shortcode though, for it will support more use cases.

**Reply by mikepianka** [reply]
Thanks, I did notice that but in my particular case I'm using embedded video as an alternative to GIF for demonstration in the docs.

**Reply by h-enk** [reply]
https://github.com/h-enk/doks/pull/244

**Reply by mikepianka** [reply]
Awesome! I'll give this a try.
